ui/plugin-system/src/components/PluginEditor/plugin-editor-api.ts

```diff
@@ -50,90 +60,103 @@ export type UsePluginEditorProps = Pick<PluginEditorProps, 'pluginType' | 'value
+
+  // Take a default kind in case user write explicitly an empty kind in the initial value
+  useEffect(() => {
+    if (value.selection.kind === '') {
+      value.selection.kind = defaultPluginKind || '';
+    }
+  }, [value.selection, defaultPluginKind]);
+
+  const { data: plugin, isFetching, error } = usePlugin(pendingSelection?.type, pendingSelection?.kind || '');
 
   useEffect(() => {
     // Nothing to do if no new plugin kind is pending
-    if (pendingKind === '') return;
+    if (!pendingSelection) return;
 
     // Can't get spec value until we have a plugin
     if (plugin === undefined) return;
 
     // Fire an onChange to change to the pending kind with initial values from the plugin
     rememberCurrentSpecState();
     onChange({
-      kind: pendingKind,
+      selection: pendingSelection,
       spec: plugin.createInitialOptions(),
     });
 
-    if (pluginType === 'Panel') {
+    if (pendingSelection.type === 'Panel') {
       const panelPlugin = plugin as PanelPlugin;
-      hideQueryState.current[pendingKind] = !!panelPlugin.hideQueryEditor;
-      if (!!panelPlugin.hideQueryEditor !== hideQueryState.current[value.kind]) {
+      hideQueryState.current[pendingSelection.kind] = !!panelPlugin.hideQueryEditor;
+      if (!!panelPlugin.hideQueryEditor !== hideQueryState.current[value.selection.kind]) {
         onHideQuery(!!panelPlugin.hideQueryEditor);
       }
     }
-
-    setPendingKind('');
-  }, [pendingKind, plugin, rememberCurrentSpecState, onChange, onHideQuery, hideQueryState, pluginType, value.kind]);
+    setPendingSelection(undefined);
+  }, [pendingSelection, plugin, rememberCurrentSpecState, onChange, onHideQuery, hideQueryState, value.selection]);
 
   /**
    * When the user tries to change the plugin kind, make sure we have the correct spec for that plugin kind before we
    * make the switch.
    */
-  const onKindChange: PluginKindSelectProps['onChange'] = (e) => {
-    const nextKind = e.target.value;
-
+  const onSelectionChange: PluginKindSelectProps['onChange'] = (nextSelection) => {
     // If we already have state for this plugin type/kind from a previous selection, just use it
-    const previousState = prevSpecState.current[pluginType]?.[nextKind];
+    const previousState = prevSpecState.current[nextSelection.type]?.[nextSelection.kind];
     if (previousState !== undefined) {
       rememberCurrentSpecState();
       onChange({
-        kind: nextKind,
+        selection: nextSelection,
         spec: previousState,
       });
     } else {
       // Otherwise, kick off the async loading process
-      setPendingKind(nextKind);
+      setPendingSelection(nextSelection);
     }
 
     if (
-      pluginType === 'Panel' &&
-      hideQueryState.current[nextKind] !== undefined &&
-      hideQueryState.current[value.kind] !== hideQueryState.current[nextKind]
+      nextSelection.type === 'Panel' &&
```

Comment:
Similar question as earlier, I’m not totally following when kind is used vs. type

I’m not against it, just curious what the guidelines are on when to use either